src/analyses/variable-sensitivity/abstract_enviroment.cpp

```diff
@@ -327,38 +302,22 @@ abstract_object_pointert abstract_environmentt::write(
 
     if(is_union)
     {
-      sharing_ptrt<union_abstract_objectt> union_abstract_object=
-        std::dynamic_pointer_cast<const union_abstract_objectt>(lhs);
-
-      sharing_ptrt<union_abstract_objectt> modified_union=
-        union_abstract_object->write_component(
-          *this, ns, remaining_stack, member_expr, rhs, merge_write);
-
-      return modified_union;
+      return lhs->write(
+        *this, ns, remaining_stack, member_expr, rhs, merge_write);
     }
     else
     {
-      sharing_ptrt<struct_abstract_objectt> struct_abstract_object=
-        std::dynamic_pointer_cast<const struct_abstract_objectt>(lhs);
-
+      // FIXME: Why does this use next_member when the union branch above
+      // FIXME: uses member_expr?
       const member_exprt next_member_expr(to_member_expr(next_expr));
```

Comment:
This looks fixed...

Reply:
Does it?

Reply:
Ah, actually I can see now that this is actually the same value in both cases, but for some reason is doing a copy here... I'll check to see why that is...